### PR TITLE
chore: refactor StringFamily::Set to use CmdArgParser

### PR DESCRIPTION
**Pull request by romange**

### CODE REVIEW

File: src/server/string_family.cc [hunk 1]
```diff
@@ -693,76 +695,83 @@ void SetCmd::RecordJournal(const SetParams& params, string_view key, string_view
...
+      sparams.flags |= SetCmd::SET_IF_EXISTS;
+    } else if (parser.Check("NX")) {
+      sparams.flags |= SetCmd::SET_IF_NOTEXIST;
+    } else {
+      string_view opt = parser.Next();
+      if (opt == "EX" || opt == "PX" || opt == "EXAT" || opt == "PXAT") {
+        int64_t int_arg = parser.Next<int64_t>();
+        if (auto err = parser.Error(); err) {
+          return builder->SendError(err->MakeReply());
+        }
```
Comment (dranikpg) [inline src/server/string_family.cc:726]: Also if you're just experimenting with the parser, the following is possible

if (parser.Peek() in {"EX", "PX", ... }) {
  // handle expiry
} else { // must be flag
  uint16_t flag = parser.Switch("GET", SetCmd::SET_GET, "STICK", SetCmd::SET_STICK....);
  flags |= flag; // returns 0 if it failed to parse, checked with Error at the end
}

if (auto err = parser.Error(); err) 
  return cntx->SendError(err->MakeReply())

Comment (romange) [reply src/server/string_family.cc]: nice. changed

File: src/server/string_family.cc [hunk 2]
```diff
@@ -693,49 +696,50 @@ void SetCmd::RecordJournal(const SetParams& params, string_view key, string_view
 }
 
 void StringFamily::Set(CmdArgList args, ConnectionContext* cntx) {
-  string_view key = ArgS(args, 0);
-  string_view value = ArgS(args, 1);
+  facade::CmdArgParser parser{args};
+
+  string_view key = parser.Next();
+  string_view value = parser.Next();
```
Comment (BorysTheDev) [inline src/server/string_family.cc:702]: you can parse 2 elements at once if you want
auto [key, value] = parser.Next<sting_view, string_view>();

File: src/server/string_family.cc [hunk 3]
```diff
@@ -693,49 +696,50 @@ void SetCmd::RecordJournal(const SetParams& params, string_view key, string_view
...
-  SinkReplyBuilder* builder = cntx->reply_builder();
+  facade::SinkReplyBuilder* builder = cntx->reply_builder();
 
-  for (size_t i = 2; i < args.size(); ++i) {
-    ToUpper(&args[i]);
+  while (parser.HasNext()) {
+    parser.ToUpper();
+    if (base::_in(parser.Peek(), {"EX", "PX", "EXAT", "PXAT"})) {
+      string_view opt = parser.Next();
+      int64_t int_arg = parser.Next<int64_t>();
```
Comment (BorysTheDev) [inline src/server/string_family.cc:713]: auto [key, value] = parser.Next<sting_view, int64_t>();

**Review by BorysTheDev — APPROVED** [review]

**Review by dranikpg — APPROVED** [review]

**Pull request merged**